### PR TITLE
Cleanup series data

### PR DESCRIPTION
**Pull request by JayKickliter**
This PR is centered around the SeriesData function, with the addition of optimizing function signatures of any of its callees.

### CODE REVIEW

File: src/outputs.cc [hunk 1]
```diff
@@ -1292,17 +1288,18 @@ void SeriesData(struct site source, struct site destination, unsigned char fresn
...
-        if ((lr.frq_mhz >= 20.0) && (lr.frq_mhz <= 100000.0) && fresnel_plot) {
-            d1 = FEET_PER_MILE * out->path.distance[x];
-            f_zone = -1.0 * sqrt(lambda * d1 * (d - d1) / d);
+        if (fresnel_plot) {
+            const double d1 = FEET_PER_MILE * out->path.distance[x];
+            f_zone = -1.0 * sqrt(wavelength * d1 * (d - d1) / d);
             fpt6_zone = f_zone * G_fzone_clearance;
         }
 
+        double r;
```
Comment (Vagabond) [inline src/outputs.cc:1297]:
```suggestion
        double r = 0.0;
```

Comment (Vagabond) [reply src/outputs.cc:1297]: and remove the else below

File: src/common.hh [hunk 2]
```diff
@@ -59,6 +59,8 @@ struct site {
     double lat;
     double lon;
     float alt;
+    /* TODO: remove the following fields. They use a huge amount of
+       stack and conflate IO with baseness logic. */
```
Comment (Vagabond) [inline src/common.hh:63]: Yes please

File: src/common.hh [hunk 3]
```diff
@@ -59,16 +59,20 @@ struct site {
...
-struct path {
+struct Path {
     std::vector<double> lat;
     std::vector<double> lon;
     std::vector<double> elevation;
     std::vector<double> distance;
-    int length;
+    ssize_t ssize() const;
+    Path() = default;
+    Path(site const &src, site const &dst);
```
Comment (Vagabond) [inline src/common.hh:75]: I don't understand this C++ stuff, what's happening here

File: src/signal-server.cc [hunk 4]
```diff
@@ -939,22 +929,13 @@ void ObstructionAnalysis(struct site xmtr, struct site rcvr, double f, FILE *out
...
-}
-
 int init(const char *sdf_path, bool debug)
 {
     // these can stay globals
     G_gpsav = 0;
     G_sdf_path[0] = 0;
     G_fzone_clearance = 0.6;
-    G_earthradius = EARTHRADIUS;
+    G_earthradius_ft = EARTHRADIUS;
```
Comment (Vagabond) [inline src/signal-server.cc:938]: Should this be EARTHRADIUS_ft? why does it even copy one constant into another?

File: src/common.hh (hunk 3 above)
Comment (Vagabond) [reply src/common.hh:75]: oh, it's a constructor!

Comment (JayKickliter) [reply src/common.hh:75]: correct. It's so the Path is "correct by construction".

File: src/signal-server.cc (hunk 4 above)
Comment (JayKickliter) [reply src/signal-server.cc:938]: I wondered the same thing. Gonna remove that global.

**Pull request merged**